matsim/src/main/java/org/matsim/core/router/NetworkRoutingInclAccessEgressModule.java

```diff
@@ -175,19 +177,27 @@ private void addBushwhackingLegFromLinkToFacilityIfNecessary(final Facility toFa
 			Leg egressLeg = populationFactory.createLeg(TransportMode.non_network_walk);
 			egressLeg.setDepartureTime(now);
 			routeBushwhackingLeg(person, egressLeg, startCoord, toFacility.getCoord(), now, startLinkId, endLinkId, populationFactory, config);
+
 			result.add(egressLeg);
-		} else if (config.plansCalcRoute().getAccessEgressWalkType().equals(AccessEgressWalkType.constantTimeToLink)) {
+		} else if (accessEgressWalkType.equals(AccessEgressWalkType.constantTimeToLink)) {
 			Leg egressLeg = populationFactory.createLeg(TransportMode.walk);
 			egressLeg.setDepartureTime(now);
 			routeBushwhackingLeg(person, egressLeg, startCoord, toFacility.getCoord(), now, startLinkId, endLinkId, populationFactory, config);
-			Id<Link> finalEndLinkId = endLinkId;
-			double egressTime = NetworkUtils.getLinkEgressTime(filteredNetwork.getLinks().get(endLinkId), mode).orElseThrow(()->new RuntimeException("Egress Time not set for link "+ finalEndLinkId.toString()));
+			double egressTime = NetworkUtils.getLinkEgressTime(egressActLink, mode).orElseThrow(()->new RuntimeException("Egress Time not set for link "+ egressActLink.getId()));
 			egressLeg.setTravelTime(egressTime);
 			egressLeg.getRoute().setTravelTime(egressTime);
 			result.add(egressLeg);
 		} else {
 			Facility fromFacility = FacilitiesUtils.wrapLink(egressActLink);
-			result.addAll(accessEgressToNetworkRouter.calcRoute(fromFacility, toFacility, now, person));
+			List<? extends PlanElement> egressTrip = accessEgressToNetworkRouter.calcRoute(fromFacility, toFacility, now, person);
+			if (this.accessEgressWalkType.equals(AccessEgressWalkType.walkToLinkPlusConstant)){
+				double accessTime = NetworkUtils.getLinkEgressTime(egressActLink,mode).orElseThrow(()->new RuntimeException("Access Time not set for link "+ egressActLink.getId().toString()));
+				Leg leg0 = TripStructureUtils.getLegs(egressTrip).get(0);
```

Comment:
Hm, this is harder than I thought. It can go wrong if there is no leg in the egress trip (->null pointer exception, ok) or that first leg is not a teleported mode, but rather has the travel time updated (e.g. setting adifferent travel time on a network mode leg has no effect, because qsim will determine the real travel time; no exception, not good). One might argue that putting the constant time in some stage activity avoids that, but that causes other problems (should be scored whereas usually stage activities are not scored). 

In the end the enum shall provide some commonly used functionality in a convenient way. The common use case is having some teleported walk for access and egress, so we should be fine in most cases. Even with walk on the network there should be a (possibly 0 m length) non_network_walk leg first (with network_walk and another non_network_walk following), so there should be no problem. 

So IMHO let's leave it like this for the time being until we have a better idea and more time. I'll try to add a comment (here in code if we have no better idea).

Thanks @jfbischoff for pushing this forward!